pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -4,16 +4,103 @@
 |-----------------|---------------------------|--------|-----------------|
 | 1A              | Working Draft             | Active | r1, 2023-07-14  |
 
-Authors:
+Authors: [@Nashatyrev], [@Menduist]
 
-Interest Group: [@vyzo]
+Interest Group: [@vyzo], [@Nashatyrev], [@Menduist]
 
 [@vyzo]: https://github.com/vyzo
+[@Nashatyrev]: https://github.com/Nashatyrev
+[@Menduist]: https://github.com/Menduist
 
 See the [lifecycle document][lifecycle-spec] for context about maturity level and spec status.
 
 [lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
 
-## Overview
+# Overview
+
+This document aims to provide a minimal extension to the [gossipsub
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to enhance the
+efficiency (minimize amplification/duplicates and decrease message latency) of
+the gossip mesh networks for larger messages. 
+
+In more specific terms, a new control message is introduced: `IDONTWANT`. It's primarily 
+intended to notify mesh peers that the node already received a message and there is no 
+need to send its duplicate.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
+
+## Parameters
+
+This section lists the configuration parameters that needs to agreed on across clients to avoid 
+ peer penalizations
+
+| Parameter                | Description                                                      | Reasonable Default |
+|--------------------------|------------------------------------------------------------------|--------------|
+| `max_idontwant_messages` | The maximum number of `IDONTWANT` messages per heartbeat per peer | ???  |
+
+
+## IDONTWANT Message
+
+### Basic scenario
+
+When the peer receives the first message instance it immediately broadcasts 
+(not queue for later piggybacking) `IDONTWANT` with the `messageId` to all its mesh peers. 
+This could be performed prior to the message validation to further increase the effectiveness of the approach.    
```

Comment:
Concerns about spam attacks triggering amplified IDONTWANT spam?

Reply:
This doesn't look like a feasible attack vector to me: 
- `IDONTWANT` is primarily intended for larger messages, so the cumulative size of resulting `IDONTWANT` messages is expected to be significantly smaller than the original message
- If an attacker is sending invalid messages to initiate `IDONTWANT` spamming it would be pretty quickly banned due to negative scoring 
- And we have `max_idontwant_messages` limit as the last resort